dynawo/sources/API/CRT/CRTCriteria.h

```diff
@@ -44,17 +44,52 @@ class Criteria {
   const boost::shared_ptr<CriteriaParams>& getParams() const;
 
   /**
-   * @brief Add an id to the component list
-   * @param id id to add
+   * @brief Add a component to the component list
+   * @param id id of the component to add
+   * @param voltageLevelId optional voltageLevelId of the component to add in case of bus criteria
    */
-  void addComponentId(const std::string& id);
+  void addComponentId(const std::string& id, const std::string& voltageLevelId = "");
 
   /**
    * @brief Add a country to the country list
    * @param id country id to add
    */
   void addCountry(const std::string& id);
 
+ private:
+  /**
+   * @class ComponentId
+   * @brief Container for component id
+   *
+   * Container for a component id and voltageLevelId.
+   */
+  class ComponentId {
```

Comment:
why not using a struct

Reply:
To cleanly encapsulate component id and voltage level id while iterating components of a criteria.
As a struct is a public class, I can use it if you want and give direct access to ids.